agent/hcp/deps.go

```diff
@@ -37,22 +38,25 @@ func NewDeps(cfg config.CloudConfig, logger hclog.Logger) (Deps, error) {
 		return Deps{}, fmt.Errorf("failed to init scada: %w", err)
 	}
 
-	metricsClient, err := client.NewMetricsClient(ctx, &cfg)
+	metricsProvider, err := NewHCPProvider(ctx, &HCPProviderCfg{})
```

Comment:
Will the `&HCPProviderCfg{}`be updated in a follow up PR, and could potentially not be nil?

Reply:
Also related to next PR, I'm assuming the sink will be started regardless of HCP config in a future PR?

Reply:
We chatted about this but to recap: We won't be ever setting `HCPProviderCfg` at startup, but I left it here as an option anyway. And yes, we will be starting the sink regardless of whether HCP is configured or not, which is why I'm always setting this as empty.

Reply:
I ended up removing this after all since I added `HCPProviderCfg` as a parameter to `Run`